packages/workspace/src/browser/quick-open-workspace.ts

```diff
@@ -75,12 +83,18 @@ export class WorkspaceQuickOpenItem extends QuickOpenItem {
     }
 
     /**
-     * Display the workspace path as part of
-     * the WorkspaceQuickOpenItem description
+     * Display shortened workspace path based on user's operating system
      * @returns workspace path
      */
     getDescription(): string {
-        return (this.workspace) ? new URI(this.workspace).path.dir.toString() : '';
+        const uri = new URI(this.workspace);
+        const path = uri.path.dir.toString();
+        const isWindows = (uri.path.root) ? Path.isDrive(uri.path.root.base) : undefined;
+
+        if (!isWindows && this.home && this.homeRegex.test(path)) {
```

Comment:
Is there a reason for using a regex, and not just looking if the string starts with `${this.home}/`?  This should work:

```ts
        if (!isWindows && this.home && path.indexOf(`${this.home}/`) === 0) {
            return path.replace(`${this.home}/`, '~/');
        }
```

It's not very likely, but if the home path contains characters that have a special meaning in regexes (e.g. `/home/[simark]`), it probably won't work correctly.  And since we don't need the extra power of regexes here, we might as well use the simple solution.